src/main/java/org/opensearch/security/support/Base64Helper.java

```diff
@@ -69,4 +71,32 @@ public static String ensureJDKSerialized(final String string) {
         // If we see an exception now, we want the caller to see it -
         return Base64Helper.serializeObject(serializable, true);
     }
+
+    /**
+     * Ensures that the returned string is custom serialized.
+     *
+     * If the supplied string is a JDK serialized representation, will deserialize it and further serialize using
+     * custom, otherwise returns the string as is.
+     *
+     * @param string original string, can be JDK or custom serialized
+     * @return custom serialized string
+     */
+    public static String ensureCustomSerialized(final String string) {
+        Serializable serializable;
+        try {
+            serializable = Base64Helper.deserializeObject(string, true);
+        } catch (Exception e) {
+            // We received an exception when de-serializing the given string. It is probably custom serialized.
```

Comment:
Can we have a check from the exception message to confirm that the cause of the exception was indeed deserialization failure using Custom serialization (this would allow to skip the check on line 91 and avoid unnecessary deserialization)

Reply:
We're trying to deserialize via JDK here. Depending on the exception message doesn't look like a good idea as it may vary across versions of JDK or a change in the underlying implementation of the `Base64JDKHelper` in future.